desk/src/components/FormControls/Grid.vue

```diff
@@ -102,28 +102,29 @@
 	</div>
 </template>
 
-<script setup>
-import { reactive, computed } from "vue"
+<script setup lang="ts">
+import { reactive, computed, PropType } from "vue"
 import { FormControl, FeatherIcon, Checkbox } from "frappe-ui"
 import Draggable from "vuedraggable"
 
 import Link from "@/components/FormControls/Link.vue"
 import IconPicker from "@/components/FormControls/IconPicker.vue"
 
 import { getRandom } from "@/utils"
+import { GridColumn, GridRow } from "@/types/controls"
 
 const props = defineProps({
 	label: {
 		type: String,
 		required: false,
 	},
 	fields: {
-		type: Array,
+		type: Array as PropType<GridColumn[]>,
 		required: true,
 	},
 })
```

Comment:
I think this can be re-written to:

```typescript
const props = defineProps<{
	fields: GridColumn[]
	label?: string,
}>()
```

Reply:
When we have default prop values, I personally don't like this syntax:

```javascript
withDefaults(defineProps<Props>, {})
```

so was avoiding this. Changed it everywhere though for consistency